File: test-automation-core/src/main/java/com/nortal/test/core/services/hooks/BeforeScenarioHook.java
package com.nortal.test.core.services.hooks;

import com.nortal.test.core.services.ScenarioContainer;

/**
 * Interface for performing before scenario preparations. Will run in @Before tag of cucumber.
 */
public interface BeforeScenarioHook {
	/**
	 * Runs pre scenario preparation operations.
	 */
	void before(ScenarioContainer scenario);
}

File: test-automation-core/src/main/kotlin/com/nortal/test/core/services/hooks/BeforeScenarioHook.kt
package com.nortal.test.core.services.hooks

import com.nortal.test.core.services.ScenarioExecutionContext

/**
 * Interface for performing before scenario preparations. Will run in @Before tag of cucumber.
 */
interface BeforeScenarioHook {
    /**
     * Runs pre scenario preparation operations.
     */
    fun before(scenario: ScenarioExecutionContext)
}